src/components/user/PeopleTable.tsx

```diff
@@ -227,184 +247,242 @@ const PeopleTable: React.FC<PeopleTableProps> = (props) => {
-        icons={tableIcons}
-        title={
-          <Typography variant="h6" component="h1">
-            {props.title}
-          </Typography>
-        }
-        page={currentPage}
-        columns={props.columns ?? columns}
-        onSelectionChange={(selectedItems, selectedItem) => {
-          // when the user wants to (un)select all items
-          // `selectedItem` will be undefined
-          if (!selectedItem) {
-            // first clear the current page because if any row was unselected
-            // the (un)select all option will select every rows
-            // which would result in duplicates
-            setSelectedParticipants((selectedParticipants) =>
-              selectedParticipants.filter(
-                ({ id }) => !currentPageIds.includes(id)
-              )
-            );
+    <Formik
+      initialValues={{
+        email: '',
+      }}
+      initialErrors={{
+        email: 'test',
+      }}
+      onSubmit={async (values, { setFieldError, setFieldValue }) => {
+        // If there is an email and it has not already been searched
+        if (values.email && !tableEmails.includes(values.email)) {
+          setLoading(true);
+          const userDetails = await getUserByEmail(values.email, api);
 
-            if (selectedItems.length > 0) {
-              setSelectedParticipants((selectedParticipants) => [
-                ...selectedParticipants,
-                ...(selectedItems.map((selectedItem) => ({
-                  id: selectedItem.id,
-                  firstname: selectedItem.firstname,
-                  lastname: selectedItem.lastname,
-                  organisation: selectedItem.organisation,
-                })) as BasicUserDetails[]),
-              ]);
-            }
+          if (!userDetails) {
+            setFieldError('email', 'Please enter valid email address');
+            setLoading(false);
+
+            return;
+          }
+
+          if (props.selectedUsers?.includes(userDetails.id)) {
+            setFieldError('email', 'User is already on the proposal');
+            setLoading(false);
 
             return;
           }
 
-          setSelectedParticipants((selectedParticipants) =>
-            (
-              selectedItem as BasicUserDetails & {
-                tableData: { checked: boolean };
+          if (!invitedUsers.some((user) => user.id === userDetails.id)) {
+            //Add users to the table
+            setInvitedUsers([userDetails].concat(invitedUsers));
+            setTableEmails(tableEmails.concat([values.email]));
+            setFieldValue('email', '');
+
+            //If we are selecting multiple users add the user as pre selected.
+            if (props.selection)
+              setSelectedParticipants(
+                selectedParticipants.concat([userDetails])
+              );
+
+            setQuery({ ...query, refreshData: !query.refreshData });
+          } else {
+            setFieldError('email', 'Could not add user to Proposal');
```

Comment:
This has lead me down a bit of a rabbit hole, invited users in this context are the users that have been added through the email invite form box. I don't _think_ this error should theoretically be every hit as the check above it should trigger as if they are in the invited user's list then they are also in the selected users this so I guess I am just going to remove it.

Saying this there are a couple of problems with the invited user list. The point of the list is that users that have been invited by the email box should appear at the top of this table else their entry in the table could be a few pages deep so it would be unclear that they have been added, which is not very user friendly. However, doing this pushes a few people off the bottom of the page and since only fetching the users on that page they will not end up appearing on the next page making them inacessable to be selected. This is quite a small bug but I am not quite sure how to fix it. 

I could just close the model when they invite the user but that then makes the task of adding multiple users very tedious. Or maybe I could muck around with the offset somehow to fetch the right users so the ones pushed off the bottom are still displayed. 

Any thoughts?

Reply:
> I could just close the model when they invite the user but that then makes the task of adding multiple users very tedious.

Yeah, that doesn't sound like a good option.

If adjusting the indexing so users pop onto the next page is feasible that sounds like it'd be a good user experience. If it's not practical, I think the next best option might be to just prepend the invited users to the first page. We might end up with a massively inflated first page compared to later ones, but at least we'd be showing all the correct data.

Reply:
Cool took me a while but got indexing right (I think), there were a couple of problems with the backend as the UOWS was returning the users out of order compared to the database. 

On another note manually extending the page size requires this work around: https://github.com/mbrn/material-table/issues/1480#issuecomment-604289636